File: src/game/events/Observable2.java
package game.events;

import java.util.HashSet;
import java.util.Set;

public class Observable2 {
    protected Set<Observer2> observers = new HashSet<>();

    public void notifyObservers(Object arg) {
        for(Observer2 o : observers)
            o.update(arg);
    }

    public void register(Observer2 o) {
        observers.add(o);
    }

    public void unregister(Observer2 o) {
        observers.remove(o);
    }

    public void unregisterAll() {
        observers.clear();
    }

    public void notify(Observer2 o, Object arg){
        o.update(arg);
    }
}

File: src/game/domain/tasks/observer/Observable2.kt
package game.domain.tasks.observer

import game.utils.dev.Log

open class Observable2 {
    @JvmField
    protected var observers: MutableSet<Observer2> = HashSet()
    fun notifyObservers(arg: Any?) {
        Log.i("notifyObservers $arg, $observers")
        for (o in observers) o.update(arg)
    }

    fun register(o: Observer2) {
        observers.add(o)
    }

    fun unregister(o: Observer2) {
        observers.remove(o)
    }

    fun unregisterAll() {
        observers.clear()
    }

    fun notify(o: Observer2, arg: Any?) {
        o.update(arg)
    }
}